From 81633afbd547b79ab2019706922c7df42aeba1ed Mon Sep 17 00:00:00 2001
From: Hongcheng Xie <hongcheng.xie@intel.com>
Date: Mon, 14 Mar 2022 16:46:17 +0800
Subject: [PATCH] Add container id into init log tag

All android container init log will transport into one kernel message,
add container id in init log tag in order to distingush different aic.

Tracked-On: OAM-101241
Signed-off-by: Hongcheng Xie <hongcheng.xie@intel.com>
Change-Id: Ic7ea1142ffc500dd761bc2aad5173b2d2621a019
---
 base/logging.cpp | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/base/logging.cpp b/base/logging.cpp
index 5bd21da66..75a6ceba0 100644
--- a/base/logging.cpp
+++ b/base/logging.cpp
@@ -19,6 +19,7 @@
 #endif
 
 #include "android-base/logging.h"
+#include "android-base/properties.h"
 
 #include <fcntl.h>
 #include <inttypes.h>
@@ -522,8 +523,19 @@ void LogMessage::LogLine(const char* file, unsigned int line, LogSeverity severi
   static auto& liblog_functions = GetLibLogFunctions();
   int32_t priority = LogSeverityToPriority(severity);
   if (liblog_functions) {
-    __android_log_message log_message = {
+    __android_log_message log_message;
+    if (tag == nullptr) {
+      std::string newTag(getprogname());
+      if (newTag == "init") {
+        std::string containerId = android::base::GetProperty("ro.container.id", "");
+        newTag += containerId;
+      }
+      log_message = {
+        sizeof(__android_log_message), LOG_ID_DEFAULT, priority, newTag.c_str(), file, line, message};
+    } else {
+      log_message = {
         sizeof(__android_log_message), LOG_ID_DEFAULT, priority, tag, file, line, message};
+    }
     liblog_functions->__android_log_write_log_message(&log_message);
   } else {
     if (tag == nullptr) {
-- 
2.25.1

